### ISSUE TITLE
Adjust dependencies for “libXScrnSaver” in RPM specification

### ISSUE THREAD

**Issue by elfring** [jitsi/jitsi#336]
[A dependency was added in the RPM specification](https://github.com/jitsi/jitsi/commit/a0e3eea89f590605ca3490f2ddf0fae078b57183 "Sysactivity notifications require libxss") for the software “libXScrnSaver”. [I noticed](http://lists.jitsi.org/pipermail/users/2017-February/012291.html "Jitsi Desktop 2.10 (and libXScrnSaver?)") that the software distribution “openSUSE” is using [the name “libXss1” for the binary package](https://build.opensuse.org/package/view_file/openSUSE:Factory/libXScrnSaver/libXScrnSaver.spec "X11 screen saver extension client library"). Now I am curious on how this situation can be resolved.
* [Can such a name become consistent in more distributions](https://lists.fedoraproject.org/archives/list/packaging@lists.fedoraproject.org/message/KAFCV23PA5J3HZFTKDYX2A6VWNRRSKGN/ "Inclusion of main version numbers in package names?")?
* Name selections are occasionally different. Would you like to [consider such details also in your RPM specification](http://lists.jitsi.org/pipermail/users/2017-February/012384.html "Jitsi Desktop 2.10 (and libXScrnSaver?)") by [corresponding checks and package name substitutions](https://en.opensuse.org/openSUSE:Build_Service_cross_distribution_howto#Handling_dependencies "openSUSE Build Service cross distribution howto: Handling dependencies")?

**Comment by elfring** [comment]
Thanks for [your adjustment of the software requirements].
With which package will this change be published?

**Comment by ibauersachs** [comment]
None yet, since the build failed.

**Comment by elfring** [comment]
Thanks for [another adjustment of the software requirements](https://github.com/jitsi/jitsi/commit/d9a52ae19507a1f1134db2af6efe901290ec97a7 "Reformatting for “Requires”").
With which files will desired changes be published?

**Comment by ibauersachs** [comment]
Build >= 2.11.5586. You see the build numbers when look at the tags on a specific commit:

![image](https://user-images.githubusercontent.com/949783/28663810-baf79e80-72be-11e7-9bba-e591534816d4.png)

**Comment by elfring** [comment]
Which package will contain software improvements also from this tag?

**Comment by ibauersachs** [comment]
The tag IS the version number of the packages.

**Comment by elfring** [comment]
* The GitHub interface does not provide a corresponding RPM file so far.
* It seems that a current RPM variant for “openSUSE” is also not listed in [the published document for download links](https://jitsi.org/downloads/ "Software download selection") at the moment.

**Comment by ibauersachs** [comment]
We don't publish binaries on Github, they are available at https://download.jitsi.org/jitsi/rpm/ or https://download.jitsi.org/jitsi/nightly/rpm/. The package is not specific to any distribution. The download page mentions RPM, not a specific distribution.

**Comment by elfring** [comment]
>… or https://download.jitsi.org/jitsi/nightly/rpm/.

I might have got an inappropriate impression from a heading like “Jitsi Desktop 2.9 Nightly Builds”.


>The package is not specific to any distribution.

I guess that you hope this more than the different name selections can show.

```
Sonne:~ # zypper install https://download.jitsi.org/jitsi/nightly/rpm/jitsi-2.11-5586.x86_64.rpm
…
Resolving package dependencies...

Problem: nothing provides libXScrnSaver needed by jitsi-2.11-5586.x86_64
…
```


>…, not a specific distribution.

I suggest to reconsider this aspect.

**Comment by ibauersachs** [comment]
Yeah, the heading is wrong.

Since `rpmbuild` apparently processes the spec file according to the host it runs on, the whole patch is useless. Well, actually, rpm seems to be a useless packaging format.

I could try using [boolean dependencies](http://rpm.org/user_doc/boolean_dependencies.html), but since they're only available since 4.13 (which was released in late 2016), this would break for you again (Suse only has 4.11 or older).

So, since we don't have a Suse build machine, this is not possible to solve for us. IMO Suse should provide libXss1 also as libXScrnSaver, which would be a simple alias for them (not the other way round, since libXScrnSaver is the standard name).

**Comment by elfring** [comment]
>…, the whole patch is useless.

The discussed software update will become useful if an appropriate build environment can be configured.
The RPM data format provides also some development challenges.


>…, this would break for you again …

I could try the tool “rpm 4.13.0.1-4.2” out on my openSUSE Tumbleweed system.


>So, since we don't have a Suse build machine, this is not possible to solve for us.

Would you like to [try the openSUSE Build Service](https://build.opensuse.org/ "Building software automatically") out?


>…, since libXScrnSaver is the standard name).

It seems that the involved distributions present different ideas for software versioning.
